### PR TITLE
implement balance warning

### PR DESCRIPTION
**Pull request by greenbigfrog**
(Somehow I forgot to switch branches)

### CODE REVIEW

File: src/discordtipbot/tipbot.cr [hunk 1]
```diff
@@ -229,6 +229,10 @@ class TipBot
     end
   end
 
+  def get_high_balance(high_balance : Int32)
+    @db.query_all("SELECT userid FROM accounts WHERE balance > $1", high_balance, &.read(Int64 | Nil))
```
Comment (z64) [inline src/discordtipbot/tipbot.cr]: Why `Int64 | Nil`? `userid` is not a nullable column, and the block should not be run if the result set is empty. Seems like you should just be using [this overload](http://crystal-lang.github.io/crystal-db/api/0.5.0/DB/QueryMethods.html#query_all%28query%2C%2Aargs%2Castype%3AClass%29-instance-method), which should return `Array(UInt64)` in your case.

File: src/discordtipbot/discordbot.cr [hunk 2]
```diff
@@ -120,6 +121,20 @@ class DiscordBot
...
+
+    # warn users that the tipbot shouldn't be used as wallet if their balance exceeds @config.high_balance
+    spawn do
+      Discord.every(6.hours) do
+        users = @tip.get_high_balance(@config.high_balance)
+        next unless users
+        next if users.nil?
+
+        users.each do |x|
+          next unless x
```
Comment (z64) [inline src/discordtipbot/discordbot.cr]: The array should never have a `nil` element.

Comment (greenbigfrog) [reply src/discordtipbot/discordbot.cr]: Compiler complained about `x` being `nil` here without that line

File: src/discordtipbot/tipbot.cr (hunk 1 above)
Comment (z64) [reply src/discordtipbot/tipbot.cr]: Yeah, looking at the [base overload implementation](https://github.com/crystal-lang/crystal-db/blob/be139c900c86e28b150b9808c74d2a4dd2bb913b/src/db/query_methods.cr#L194-L206) for `query_all`, it should always return an array, not `Array(U) | Nil`:
```cr
    def query_all(query, *args, &block : ResultSet -> U) : Array(U) forall U
      ary = [] of U
      query_each(query, *args) do |rs|
        ary.push(yield rs)
      end
      ary
    end
```

Comment (greenbigfrog) [reply src/discordtipbot/tipbot.cr]: Cool. Somehow I missed that overload.

**Pull request merged**